### ISSUE TITLE
Cannot assign function to object

### ISSUE THREAD

**Issue by netei** [eventualbuddha/decaffeinate#82]
take this file : 

```
t={}
t.n=()->console.log 1
```
Outputs this : 

```
TypeError: Cannot read property 'data' of undefined
    at isStaticMethod (.nvm/versions/node/v0.12.7/lib/node_modules/decaffeinate/dist/decaffeinate.cjs.js:702:154)
    at patchFunctionStart (.nvm/versions/node/v0.12.7/lib/node_modules/decaffeinate/dist/decaffeinate.cjs.js:4143:11)
    at .nvm/versions/node/v0.12.7/lib/node_modules/decaffeinate/dist/decaffeinate.cjs.js:4891:5
    at traverse (.nvm/versions/node/v0.12.7/lib/node_modules/decaffeinate/dist/decaffeinate.cjs.js:473:23)
    at .nvm/versions/node/v0.12.7/lib/node_modules/decaffeinate/dist/decaffeinate.cjs.js:464:11
    at Array.forEach (native)
    at .nvm/versions/node/v0.12.7/lib/node_modules/decaffeinate/dist/decaffeinate.cjs.js:462:15
    at Array.forEach (native)
    at descend (.nvm/versions/node/v0.12.7/lib/node_modules/decaffeinate/dist/decaffeinate.cjs.js:459:32)
    at .nvm/versions/node/v0.12.7/lib/node_modules/decaffeinate/dist/decaffeinate.cjs.js:4902:5
```

**Comment by netei** [comment]
Updating to latest version makes this work correctly

**Comment by netei** [comment]
Updating to latest version makes this work correctly

**Comment by eventualbuddha** [comment]
Yep, this was fixed by e0257f07dc8a3c335815588c55d9e116e18c04b8 in v1.40.17.
